### PR TITLE
refactor!: Seed filter container policy

### PR DESCRIPTION
**Pull request by CouthuresJeremy**
Refactor the seed filter to allow other container types than std::vector.
Using a container policy allows to keep the functions virtual.

### CODE REVIEW

File: Core/include/Acts/Seeding/ContainerPolicy.hpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include <iterator>
+#include <vector>
+
+#include <stdlib.h>
```
Comment (andiwand) [inline Core/include/Acts/Seeding/ContainerPolicy.hpp:14]:
```suggestion
```

**Review by andiwand — COMMENTED** [review]
@LuisFelipeCoelho @CarloVarni may want to have a look at this

File: Examples/Algorithms/TrackFinding/src/SeedingAlgorithm.cpp [hunk 2]
```diff
@@ -312,9 +312,11 @@ ActsExamples::ProcessCode ActsExamples::SeedingAlgorithm::execute(
...
 
   for (const auto [bottom, middle, top] : spacePointsGrouping) {
-    m_seedFinder.createSeedsForGroup(
-        m_cfg.seedFinderOptions, state, spacePointsGrouping.grid(),
-        std::back_inserter(seeds), bottom, middle, top, rMiddleSPRange);
+    VectorPolicy out_policy_cont(seeds);
+    GenericBackInserter back_inserter(out_policy_cont);
+    m_seedFinder.createSeedsForGroup(m_cfg.seedFinderOptions, state,
+                                     spacePointsGrouping.grid(), back_inserter,
+                                     bottom, middle, top, rMiddleSPRange);
```
Comment (AJPfleger) [inline Examples/Algorithms/TrackFinding/src/SeedingAlgorithm.cpp:319]:
```suggestion
    VectorPolicy outPolicyContainer(seeds);
    GenericBackInserter backInserter(outPolicyContainer);
    m_seedFinder.createSeedsForGroup(m_cfg.seedFinderOptions, state,
                                     spacePointsGrouping.grid(), backInserter,
                                     bottom, middle, top, rMiddleSPRange);
```
These sections of ACTS do not really align with our [naming convention](https://acts.readthedocs.io/en/v34.0.0/codeguide.html#naming), but I had the feeling that new parts were usually added in ACTS-style. Maybe you could adapt this (for YOUR changes only) and only keep the templated paramaters like `template <typename sp_range_t>` in snake_case.

Comment (CouthuresJeremy) [reply Examples/Algorithms/TrackFinding/src/SeedingAlgorithm.cpp]: It should be fine now, thanks!

**Comment by CarloVarni** [comment]
lgtm

**Comment by CarloVarni** [comment]
I believe this is a breaking PR

**Comment by CouthuresJeremy** [comment]
Is there any remaining issue with this pull request @CarloVarni @LuisFelipeCoelho?

**Comment by CarloVarni** [comment]
Hi @CouthuresJeremy with c++20 this would be easily implementable without introducing a container policy. What about we rethink about this by using some concepts? 

@andiwand @AJPfleger what do you think?

**Comment by paulgessinger** [comment]
C++20 is in the pipeline now.

**Comment by CarloVarni** [comment]
https://github.com/acts-project/acts/pull/336 has address this. This PR has been superseeded and can be closed

**Pull request closed without merging**